### PR TITLE
Update component state API to work with account components and add a test

### PR DESCRIPTION
**Pull request by LukasGasior1**

### CODE REVIEW

File: core-rust/core-api-server/src/core_api/helpers.rs [hunk 1]
```diff
@@ -33,17 +33,31 @@ pub(crate) fn read_known_substate(
     node_module_id: NodeModuleId,
     substate_offset: &SubstateOffset,
 ) -> Result<PersistedSubstate, ResponseError<()>> {
+    read_optional_substate(state_manager, renode_id, node_module_id, substate_offset).ok_or_else(
+        || {
+            MappingError::MismatchedSubstateId {
+                message: format!(
+                    "Substate {:?} not found under {:?}",
```
Comment (jakrawcz-rdx) [inline core-rust/core-api-server/src/core_api/helpers.rs:40]: (drive-by) let's also log `node_module_id`?

File: core-rust/core-api-server/src/core_api/handlers/state_component.rs [hunk 2]
```diff
@@ -49,42 +51,69 @@ fn handle_state_component_internal(
...
             state_manager,
             RENodeId::GlobalComponent(component_address),
             NodeModuleId::SELF,
             &substate_offset,
-        )?;
-        let PersistedSubstate::ComponentState(substate) = loaded_substate else {
+        );
+        if let Some(PersistedSubstate::ComponentState(substate)) = loaded_substate_opt {
+            Some(substate)
+        } else if let Some(..) = loaded_substate_opt {
```
[the remark this answers is not in the document]
Comment (jakrawcz-rdx) [reply core-rust/core-api-server/src/core_api/handlers/state_component.rs:62]: _(plz disregard if it does not help with readability)_

File: core/src/test/java/com/radixdlt/rev2/coreapi/REv2ComponentStateApiTest.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+        // Assert: verify that the request succeeds and returns a response
+        assertEquals("Account", componentStateResponse.query("/info/blueprint_name"));
+      } finally {
+        apiServer.stop();
+      }
+    }
+  }
+
+  private static JSONObject postJson(int coreApiPort, String path, JSONObject request)
```
[the remark this answers is not in the document]
Comment (LukasGasior1) [reply core/src/test/java/com/radixdlt/rev2/coreapi/REv2ComponentStateApiTest.java:211]: that was my ignorance...:D I can totally reuse it here (and the `DeterministicCoreApiTestBase` which I've overlooked)

**Review by jakrawcz-rdx — APPROVED** [review]
